ext/nnstreamer/tensor_filter/tensor_filter_qnn.cc

```diff
@@ -0,0 +1,680 @@
+  if (mrt != MODEL_NO_ERROR) {
+    nns_loge ("Failed to compose graphs. Error: %d", (int) mrt);
+    return false;
+  }
+
+  /** @todo Support multiple graphs. Assume graph_count == 1 for now. */
+  if (graph_count != 1U) {
+    nns_loge ("Only support single graph for now. graph_count: %u", graph_count);
+    return false;
+  }
+
+  GraphInfo_t *graph_info = graphs_info_[0];
+
+  /* Parse input tensors */
+  uint32_t num_input_tensors = graph_info->numInputTensors;
+  input_tensors_.reserve (num_input_tensors);
+  for (uint32_t j = 0; j < num_input_tensors; ++j) {
+    Qnn_Tensor_t _t = QNN_TENSOR_INIT;
+    if (!deepCopyQnnTensor (&_t, &graph_info->inputTensors[j])) {
+      nns_loge ("Failed to copy tensor info.");
+      return false;
+    }
+    _t.v1.clientBuf.dataSize = qnnTensorDataSize (&_t);
+    input_tensors_.push_back (_t);
+  }
+
+  /* Parse output tensors */
+  uint32_t num_output_tensors = graph_info->numOutputTensors;
+  output_tensors_.reserve (num_output_tensors);
+  for (uint32_t j = 0; j < num_output_tensors; ++j) {
+    Qnn_Tensor_t _t = QNN_TENSOR_INIT;
+    if (!deepCopyQnnTensor (&_t, &graph_info->outputTensors[j])) {
+      nns_loge ("Failed to copy tensor info.");
+      return false;
+    }
+    _t.v1.clientBuf.dataSize = qnnTensorDataSize (&_t);
+    output_tensors_.push_back (_t);
+  }
+
+  /* Finalize Graphs */
+  if (QNN_GRAPH_NO_ERROR
+      != qnn_interface_.graphFinalize (graph_info->graph, nullptr, nullptr)) {
+    nns_loge ("Failed to finalize QNN graph");
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * @brief Free QnnManager resources.
+ */
+void
+QnnManager::fini ()
+{
+  auto _free_qnn_tensor = [] (Qnn_Tensor_t &tensor) {
+    free ((void *) tensor.v1.name);
+    free (tensor.v1.dimensions);
+    if (tensor.version == QNN_TENSOR_VERSION_2) {
+      if (tensor.v2.isDynamicDimensions) {
+        free (tensor.v2.isDynamicDimensions);
+      }
+    }
+  };
+
+  /* release qnn tensors */
+  for (auto &t : input_tensors_) {
+    _free_qnn_tensor (t);
+  }
+  for (auto &t : output_tensors_) {
+    _free_qnn_tensor (t);
+  }
+
+  input_tensors_.clear ();
+  output_tensors_.clear ();
+
+  /* free graphs from model */
+  if (model_module) {
+    FreeGraphsInfoFn *free_graphs;
+    if (!g_module_symbol (model_module, "QnnModel_freeGraphsInfo", (void **) &free_graphs)) {
+      nns_logw ("Failed to resolve symbol 'QnnModel_freeGraphsInfo' from model: %s",
+          g_module_error ());
+    } else {
+      free_graphs (&graphs_info_, 1U);
+    }
+
+    g_module_close (model_module);
+  }
+
+  /* release context */
+  if (backend_module) {
+    if (nullptr != qnn_interface_.contextFree) {
+      qnn_interface_.contextFree (context_handle_, nullptr);
+    }
+
+    /* release backend */
+    if (nullptr != qnn_interface_.backendFree) {
+      qnn_interface_.backendFree (backend_handle_);
+    }
+
+    g_module_close (backend_module);
```

Comment:
line 410 and 424, let's clearly set null after close module
(backend_module = nullptr;)

Reply:
Those lines are added, PTAL.